### ISSUE TITLE
always_reload is not working on invoice and shipping addresses 

### ISSUE THREAD

**Issue by themreza** [odoo/odoo#58711]
Impacted versions:
I've tested this on Odoo 13.0 Community Edition 

Steps to reproduce:
1. Inherit `sale.view_order_form` and add the built-in contexts `'show_address': 1, 'address_inline': 1` to fields `partner_invoice_id` and `partner_shipping_id`. This displays the partner / customer address next to their name on the dropdown labels in sale view order.

```
<?xml version="1.0" encoding="utf-8"?>
<odoo>
    <data>
          <record id="view_order_form_partner_dropdown_info" model="ir.ui.view">
                    <field name="name">sale.order.form.partner.dropdown.info</field>
                    <field name="model">sale.order</field>
                    <field name="inherit_id" ref="sale.view_order_form"/>
                    <field name="arch" type="xml">
                        <field name="partner_invoice_id" position="replace">
                            <field name="partner_invoice_id" groups="sale.group_delivery_invoice_address" context="{'default_type':'invoice', 'show_address': 1, 'address_inline': 1}" options='{"always_reload": True}'/>
                        </field>
                        <field name="partner_shipping_id" position="replace">
                            <field name="partner_shipping_id" groups="sale.group_delivery_invoice_address" context="{'default_type':'delivery', 'show_address': 1, 'address_inline': 1}" options='{"always_reload": True}'/>
                        </field>
                    </field>
          </record>
    </data>
</odoo>

```

2. Enable _Customer Addresses (Select specific invoice and delivery addresses)_ in Settings > Sales

3. Create a new quotation, choose a customer, and observe that although the invoice and shipping addresses have the inline addresses, the automatic selection of their values does not result display the correct labels.

Current behavior:
The address fields only show the record name of the partner / customer, without going through `name_get` processing.

Expected behavior:
Since `always_reload` is set to `True` on the fields, the automatically-selected addresses should display the inline-addresses as well.

The code responsible for automatically changing the address fields is: https://github.com/odoo/odoo/blob/13.0/addons/sale/models/sale.py#L372

This is related to these issues which were fixed, but I'm still facing the bug:
https://github.com/odoo/odoo/issues/42444
https://github.com/odoo/odoo/pull/42401/files

Video/Screenshot link (optional):
https://streamable.com/u64xha

**Comment by pedrobaeza** [comment]
Do you have latest code? Is this reproducible on runbot?

**Comment by themreza** [comment]
Yes, I'm running the latest code.

Here is a demo on the latest runbot v13.0 CE:
https://streamable.com/ni62q0

**Comment by pedrobaeza** [comment]
I don't think this is an error as such, as the name_get is done at other level, but I let this opened for other to judge.

**Comment by themreza** [comment]
How do you suggest to fix this behavior? I'd like to have the label displayed with the inline address after the partner field has changed.

**Comment by pedrobaeza** [comment]
I don't think that's possible right now that way.

**Comment by Yenthe666** [comment]
To this date it is still a limitation upwards to Odoo V17 I'm afraid :)
It is debatable but it would for sure be handy to be able to do it.
